### ISSUE TITLE
Android/Arm cross-compiler no longer build with -CfVFPV3

### ISSUE THREAD

**Issue by michaliskambi** [LongDirtyAnimAlf/fpcupdeluxe#460]
Past versions of FpcUpDeluxe were making Android/Arm cross-compiler with `-CfVFPV3`, which was very good for my use-case. We need "hard floats" to run Castle Game Engine applications on Android with reasonable speed (see https://github.com/castle-engine/castle-engine/wiki/Android-FAQ#notes-about-compiling-with-hard-floats--cfvfpv3 ).

However, since some time (I'm not sure exactly when, I'm afraid) it seems FpcUpDeluxe by default makes Android/Arm cross-compiler without `-CfVFPV3`. It can be adjusted in the settings of course, but it's still a bother -- and a potential problem to new users, not familiar with Android or FPC cross-compilers etc. We ideally want to direct such users to FpcUpDeluxe and say "use it, it makes good cross-compilers by default" but instead now we have to document that you need to adjust Android/Arm settings to add `-CfVFPV3`, like we do here: https://github.com/castle-engine/castle-engine/wiki/Android-SDK-and-NDK-troubleshooting#compilation-errors-trying-to-use-a-unit-which-was-compiled-with-a-different-fpu-mode .

Was this a simple mistake? Can you restore using `-CfVFPV3` for Android/Arm cross-compiler? Or was just a deliberate decision, and we just have to adjust our CGE docs to account for this?

**Comment by LongDirtyAnimAlf** [comment]
Could you please elaborate on this.
As far as fpcupdeluxe is concerned, it is (or should) still uses (at least on my latest vanilla Win10 install):

> (OPT: -dFPC_ARMHF) [CROSSOPT: -CfVFPV3 -OoFASTMATH -CaEABI]

**Comment by michaliskambi** [comment]
Ah, sorry --- I should have written that I'm using Linux/x86_64 as a host.

Indeed it works fine when the host is Windows, I see the options you show (including `-CfVFPV3`) in the output.

But when the host is Linux, they are not present. The output is

```
Trying to build the cross-compiler for android-arm.
Got all tools. Started to build cross-compiler.
Fpcupdeluxe: FPC cross-builder: Building compiler for android-arm.
fpcupdeluxe: info: FPC arm-android cross-builder: Detected source version FPC (source): 3.2.3
fpcupdeluxe: info: FPC arm-android cross-builder: Using compiler with version: 3.2.3
fpcupdeluxe: Done !!
```

I attach the full log. So it looks like the bug is not present on Windows. I just tested with latest v2.2.0e version of fpcupdeluxe.

[fpcupdeluxe.txt](https://github.com/LongDirtyAnimAlf/fpcupdeluxe/files/7569532/fpcupdeluxe.txt)

**Comment by LongDirtyAnimAlf** [comment]
Could it be that you have left-over options in your install directory ?
A vanilla install on a Linux machine behaves as expected !

![crossarm](https://user-images.githubusercontent.com/10783972/142627891-31eb27e6-4e47-44a6-81d4-c0237116ff57.JPG)

**Comment by michaliskambi** [comment]
> Could it be that you have left-over options in your install directory ?

That is indeed what happened. I don't know why, but `fpcupdeluxe.ini` in my FPC/Lazarus installation had empty `CrossBuildOptions` for `[arm-android]` (so, without `-CfVFPV3`). That is also why cleaning / upgrading fpcupdeluxe directory wasn't helping me -- it was always using my incorrect `fpcupdeluxe.ini` in FPC/Lazarus installation, outside of the directory where I keep fpcupdeluxe exe.

I checked now by making an install to a new FPC/Lazarus directory, and everything is good, FPC units are built with `-CfVFPV3` (for Arm/Android) and CGE applications can be built with `castle-engine --target=android package` out-of-the-box.

Sorry for the noise, and thanks for the help -> all is good :)

I have simplified the instructions on https://github.com/castle-engine/castle-engine/wiki/fpcupdeluxe accordingly.
